app/controllers/avalon_marker_controller.rb

```diff
@@ -0,0 +1,135 @@
+  def show
+    lookup_marker
+    render json: @marker.pretty_annotation
+  end
+
+  # Creates a marker and renders it as JSON
+  # @param [Hash] params the parameters used for creating a marker
+  # @option params [String] :master_file the pid of the MasterFile to be used for this marker's source
+  # @option params [String] :playlist_item the pid of the PlaylistItem to which this marker belongs
+  # @option params [String] :title the title to use for the marker, defaults the the MasterFile title
+  # @option params [String] :start_time the time point the marker begins, defaults to 0
+  # @example Rails Console command to create a marker based of the MasterFile whose pid is avalon:20 and default values for all other fields in the marker
+  #    app.post('/avalon_marker/', {master_file: 'avalon:20'})
+  def create
+    marker_params = params['avalon_marker']
+    render json: { message: 'Masterfile Not Supplied' }, status: 400 and return if marker_params[:master_file].nil?
+    render json: { message: 'Playlist Item Not Supplied' }, status: 400 and return if marker_params[:playlist_item].nil?
+    mf = MasterFile.where(id: marker_params[:master_file])[0]
+    render json: { message: @not_found_messages[:master_file] }, status: 400 and return if mf.nil?
+    pi = PlaylistItem.where(id: marker_params[:playlist_item])[0]
+    render json: { message: @not_found_messages[:playlist_item] }, status: 400 and return if pi.nil?
+
+    @marker = AvalonMarker.create(master_file: mf, playlist_item: pi)
+    selected_key_updates
+
+    if @marker.persisted?
+      render json: { id: @marker.id, marker: {title: @marker.title, start_time: @marker.start_time}, message: "Add marker to playlist item was successful." }, status: 201 and return
+    else
+      render json: { message: @marker.errors.full_messages }, status: 400 and return
+    end
+  end
+
+  # Updates a marker and renders it as JSON
+  # @param [Hash] params the parameters used for updating a marker
+  # @option params [String] :id the id of the marker to update
+  # @option params [String] :title the title to use for the marker
+  # @option params [String] :start_time the time point the marker
+   # @example Rails Console command to update the title of the marker with a uuid of 56 to be 'Hail'
+  #    app.put('/avalon_marker/56', {title: 'Hail'})
+  def update
+    lookup_marker
+    selected_key_updates
+    render json: {id: @marker.id, marker: @marker.pretty_annotation}
+  end
+
+  # Destroy a marker based on uuid
+  # @param [Hash] params the parameters used by the controller
+  # @option params [String] :id the uuid of the marker you wish to destroy
+  # @example Rails Console command to destroy the marker with an uuid of 56
+  #    app.delete('/avalon_marker/56')
+  def destroy
+    lookup_marker
+    @marker.destroy
+    render json: { action: 'destroy', id: @marker.id, success: true }
+  end
+
+  # Looks up a marker using the id key in params and sets @marker
```

Comment:
The methods below destroy should probably be private.  (lookup_marker, selected_key_updates, not_found)